### PR TITLE
mcs: update node every restart

### LINKED ISSUE #8154: pdms: tiup playground restart cluster failed
### What did you do?
Using the following command start tidb cluster with tag.
Then exit it and run it again as followings.
```
tiup playground v8.1.0-pre --tag test-v8.1.0-pdms-2 --pd.mode ms
Start pd instance:v8.1.0-pre
Start tso instance:v8.1.0-pre
Start scheduling instance:v8.1.0-pre
Start tikv instance:v8.1.0-pre
Start tidb instance:v8.1.0-pre
Waiting for tidb instances ready
[127.0.0.1:4000](http://127.0.0.1:4000/) ... Done
Start tiflash instance:v8.1.0-pre
Waiting for tiflash instances ready
[127.0.0.1:3930](http://127.0.0.1:3930/) ... Done

🎉 TiDB Playground Cluster is started, enjoy!

Connect TiDB:    mysql --comments --host [127.0.0.1](http://127.0.0.1/) --port 4000 -u root
TiDB Dashboard:  http://127.0.0.1:44437/dashboard
Grafana:         http://127.0.0.1:3000/
receive command: scale-out
Start tso instance:v8.1.0-pre
receive command: scale-out
Start tso instance:v8.1.0-pre
receive command: scale-out
Start tso instance:v8.1.0-pre
receive command: scale-out
Start scheduling instance:v8.1.0-pre
receive command: scale-out
Start scheduling instance:v8.1.0-pre
receive command: scale-out
Start scheduling instance:v8.1.0-pre
receive command: display
^CGot signal interrupt (Component: playground ; PID: 1196454)
Playground receive signal:  interrupt
Wait tiflash(1196986) to quit...
Wait prometheus(1196945) to quit...
Wait ng-monitoring(1196946) to quit...
Wait grafana(1196974) to quit...
Grafana quit
prometheus quit
ng-monitoring quit
tiflash quit
Wait tidb(1196538) to quit...
tidb quit
Wait tikv(1196523) to quit...
tikv quit
Wait pd(1196467) to quit...
pd quit
Wait tso(1196478) to quit...
tso quit
Wait tso(1198603) to quit...
tso quit
Wait tso(1198612) to quit...
tso quit
Wait tso(1198624) to quit...
tso quit
Wait scheduling(1196490) to quit...
scheduling quit
Wait scheduling(1198680) to quit...
scheduling quit
Wait scheduling(1198691) to quit...
scheduling quit
Wait scheduling(1198701) to quit...
scheduling quit
[root@hongmei ~]# tiup playground v8.1.0-pre --tag test-v8.1.0-pdms-2 --pd.mode ms
Start pd instance:v8.1.0-pre
Start tso instance:v8.1.0-pre
Start scheduling instance:v8.1.0-pre
Start tikv instance:v8.1.0-pre
```

### What did you see instead?
```
[root@hongmei ~]# tiup playground v8.1.0-pre --tag test-v8.1.0-pdms-1 --pd.mode ms
Start pd instance:v8.1.0-pre
Start tso instance:v8.1.0-pre
Start scheduling instance:v8.1.0-pre
Start tikv instance:v8.1.0-pre
Force pd(1192380) to quit...
pd quit
Force tso(1192392) to quit...
tso quit
Force scheduling(1192404) to quit...
scheduling quit
Error: Playground bootstrapping failed: error check PD Micro Service's TSO health, operation timed out after 5m0s
[root@hongmei ~]# tiup playground --version
tiup version 1.15.0 tiup
Go Version: go1.21.0
Git Ref: fix-health-check
GitHash: c08c7dbf
```

### What version of PD are you using (`pd-server -V`)?
v8.1.0-pre

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #8154

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/keyspace/tso_keyspace_group.go [hunk 1]
```diff
@@ -792,10 +792,10 @@ func (m *GroupManager) AllocNodesForKeyspaceGroup(id uint32, desiredReplicaCount
 			if addr == "" {
```
Comment (lhy1024) [inline pkg/keyspace/tso_keyspace_group.go:792]: Do we need to check whether tso node is available here?

Comment (rleungx) [reply pkg/keyspace/tso_keyspace_group.go:792]: The node balancer will be updated through the watch event.

Comment (lhy1024) [reply pkg/keyspace/tso_keyspace_group.go:792]: When a watch event is delayed, an unavailable tso is allocated. Maybe we can add an available check?

File: pkg/keyspace/tso_keyspace_group.go [hunk 2]
```diff
@@ -194,23 +191,26 @@ func (m *GroupManager) allocNodesToAllKeyspaceGroups(ctx context.Context) {
...
+				if exist, addr := m.IsExistNode(member.Address); exist {
+					existMembers[addr] = struct{}{}
+				}
+			}
+			numExistMembers := len(existMembers)
+			if numExistMembers != 0 && numExistMembers == len(group.Members) && numExistMembers == m.GetNodesCount() {
+				continue
+			}
+			if numExistMembers < utils.DefaultKeyspaceGroupReplicaCount {
+				nodes, err := m.AllocNodesForKeyspaceGroup(group.ID, existMembers, utils.DefaultKeyspaceGroupReplicaCount)
```
Comment (lhy1024) [inline pkg/keyspace/tso_keyspace_group.go:206]: If numExistMembers is 1 and len(nodes) also is 1, will it always enter this branch and return in L790?

**Review by JmPotato — APPROVED** [review]

File: pkg/keyspace/tso_keyspace_group.go (hunk 2 above)
Comment (rleungx) [reply pkg/keyspace/tso_keyspace_group.go:206]: No, it will continue if m.GetNodesCount() also is 1.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**